doc/hotspot-style.md

```diff
@@ -703,6 +701,274 @@ Some relevant sections from cppreference.com:
 Although related, the use of `std::initializer_list` remains forbidden, as
 part of the avoidance of the C++ Standard Library in HotSpot code.
 
+### Local Function Objects
+
+* Local function objects, including lambda expressions, may be used.
+* Lambda expressions must only be used as a downward value.
+* Prefer `[&]` as the capture list of a lambda expression.
+* Return type deduction for lambda expressions is permitted, and indeed encouraged.
+* An empty parameter list for a lambda expression may be elided.
+* A lambda expression must not be `mutable`.
+* Generic lambda expressions are permitted.
+* Lambda expressions should be relatively simple.
+* Anonymous lambda expressions should not overly clutter the enclosing expression.
+* An anonymous lambda expression must not be directly invoked.
+* Bind expressions are forbidden.
+
+Single-use function objects can be defined locally within a function.  This
+is an alternative to having a function or function object class defined at
+class or namespace scope.  Instead, a function class definition is placed
+directly at the point of use.
```

Comment:
Suggested rewording:

> ... namespace scope, instead placing it directly at the point of use.

Reply:
The wording was intentional; "it" would refer to "a function or function object class", but a *function* can't be locally defined.  But it should be consistent about "function object class" vs "function class".  The former is the terminology used by the Standard.

Reply:
There should be a consistent subject for each of those sentences, such that "it" would unambiguously be that subject, and then you can combine the second and third sentences into one.

Reply:
Changing the paragraph to this is probably better.  The third sentence didn't really add anything.

"Single-use function objects can be defined locally within a function, directly at the point of use.  This is an alternative to having a function or function object class defined at class or namespace scope."